src/mono/Directory.Build.props

```
@@ -40,10 +40,18 @@
   <!-- Directory to provision and use emscripten if EMSDK_PATH env variable is not set -->
   <PropertyGroup Condition="'$(TargetsBrowser)' == 'true'">
     <ProvisionEmscriptenDir>$([MSBuild]::NormalizeDirectory('$(MSBuildThisFileDirectory)', 'wasm', 'emsdk'))</ProvisionEmscriptenDir>
-    <ShouldProvisionEmscripten Condition="'$(OS)' == 'Windows_NT' and '$(TargetArchitecture)' == 'wasm' and '$(EMSDK_PATH)' == '' and !Exists('$(ProvisionEmscriptenDir)')">true</ShouldProvisionEmscripten>
+    <ShouldProvisionEmscripten Condition="'$(EMSDK_PATH)' == '' and !Exists('$(ProvisionEmscriptenDir)')">true</ShouldProvisionEmscripten>
     <EMSDK_PATH Condition="Exists('$(ProvisionEmscriptenDir)') and '$(EMSDK_PATH)' == ''">$(ProvisionEmscriptenDir.Replace('\', '/'))</EMSDK_PATH>
   </PropertyGroup>
 
+  <!-- Directory to provision and use WASI sdk if WASI_SDK_PATH env variable is not set -->
+  <PropertyGroup Condition="'$(TargetsWasi)' == 'true'">
+    <ProvisionWasiSdkDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'wasi-sdk'))</ProvisionWasiSdkDir>
```

nit: `bin` is for actual output binaries, we should put the wasi-sdk into `ArtifactsObjDir` instead
```suggestion
    <ProvisionWasiSdkDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsObjDir)', 'wasi-sdk'))</ProvisionWasiSdkDir>
```